.circleci/config/commands/check-branch-name.yml

```diff
@@ -0,0 +1,12 @@
+description: >
+  Check if branch name starts with ui/ or docs/ and if so, exit.
+steps:
+  - run:
+      name: Check branch name
+      command: |
+        # If the branch being tested starts with ui/ or docs/ we want to exit stop the job without failing
+        [[ "$CIRCLE_BRANCH" = ui/* || "$CIRCLE_BRANCH" = docs/* ]] && {
+          # stop the job from this step
+          circleci-agent step halt
+          exit 0
+        }
```

Comment:
exit 0 required for non-matching too (see https://circleci.com/gh/hashicorp/vault/42229 where it otherwise exits with status 1)